src/character.h

```diff
@@ -1371,15 +1371,16 @@ class Character : public Creature, public visitable
         void mod_cost_timer( const trait_id &mut, time_duration mod );
 
         /** Picks a random valid mutation and gives it to the Character, possibly removing/changing others along the way */
-        void mutate( const int &true_random_chance, bool use_vitamins );
+        void mutate( const int &true_random_chance, bool use_vitamins, double ( *rng )( double,
```

Comment:
Please don't call this rng, you're shadowing the rng function in a way that's pretty opaque in ~200 line function.
```suggestion
        void mutate( const int &true_random_chance, bool use_vitamins, double ( *rng_fn )( double,
```
Also, you should be able to use a std::function here instead of a function pointer.

Reply:
Thank you for your input but I decided to undo that attempt, I don't think I want to mess with things like that. I reverted this PR back to just fixing the current test instead of replacing it.